packages/api/src/router/absence.ts

```diff
@@ -6,14 +7,17 @@ export const absenceRouter = createTRPCRouter({
   createAbsence: protectedProcedure
     .input(
       z.object({
-        participantId: z.number().nonnegative(),
         absenceDate: z.date(),
       }),
     )
     .mutation(async ({ input, ctx }) => {
+      const user = await ctx.db.user.findFirst({ where: { id: ctx.userId } });
+      if (!user) {
+        throw new TRPCError({ code: "NOT_FOUND" });
+      }
       await ctx.db.absence.create({
         data: {
-          participantId: input.participantId,
+          participantId: user.participantId,
           absenceDate: input.absenceDate,
         },
       });
```

Comment:
should we make this an upsert instead (not related to ticket but might be necessary)

Reply:
do we use upsert so that if they find a record that already has the `absenceDate` and `participantId`, then it would do nothing?

Reply:
yep -- I think that if we don't use an upsert, then there will be an error if we try to create one that already exists

Reply:
:o ok thank you!